lib/src/rules/use_setters_to_change_a_property.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.use_setters_to_change_a_property;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/standard_resolution_map.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+const _desc =
+    r'Use a setter for operations that conceptually change a property.';
+
+const _details = r'''
+
+**DO** use a setter for operations that conceptually change a property.
+
+**BAD:**
+```
+rectangle.setWidth(3);
+button.setVisible(false);
+```
+
+**GOOD:**
+```
+rectangle.width = 3;
+button.visible = false;
+```
+
+''';
+
+class UseSettersToChangeAProperty extends LintRule {
+  _Visitor _visitor;
+  UseSettersToChangeAProperty()
+      : super(
+            name: 'use_setters_to_change_a_property',
+            description: _desc,
+            details: _details,
+            group: Group.style) {
+    _visitor = new _Visitor(this);
+  }
+
+  @override
+  AstVisitor getVisitor() => _visitor;
+}
+
+class _Visitor extends SimpleAstVisitor {
+  final LintRule rule;
+  _Visitor(this.rule);
+
+  @override
+  visitMethodDeclaration(MethodDeclaration node) {
+    if (node.isStatic ||
+        node.isSetter ||
+        resolutionMap.elementDeclaredByMethodDeclaration(node).isOverride ||
+        node.parameters?.parameters?.length != 1 ||
+        (node.returnType != null && node.returnType.type.name != 'void')) {
+      return;
+    }
+    final body = node.body;
+    if (body is BlockFunctionBody && body.block.statements.length == 1) {
```

Comment:
I agree that the second case should be a separate diagnostic, though I'd lean toward making it a hint because it's never useful to return a value from a `void` method. But the first is a common short-hand used for methods that will fit on a single line.

But I'm fine with not handling it for now and waiting to l see what users expect.